executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -0,0 +1,390 @@
+#![allow(unused)]
+use std::{
+    collections::{BTreeSet, HashSet},
+    fmt::Display,
+};
+
+use bit_vec::BitVec;
+use powdr_ast::analyzed::{AlgebraicReference, Identity};
+use powdr_number::FieldElement;
+
+use crate::witgen::{machines::MachineParts, FixedData};
+
+use super::{
+    affine_symbolic_expression::Effect,
+    cell::Cell,
+    witgen_inference::{FixedEvaluator, Variable, WitgenInference},
+};
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug)]
+/// A variable for the block machine solver.
+enum CellOrArg {
```

Comment:
Could we avoid the 'Or' in this name by calling this enum `Variable` instead of the trait?